Autopilot/Inc/SensorFusion.hpp

```diff
@@ -88,37 +89,37 @@ void SF_Init(void);
 /**
  * Generates fused sensor data. Should be called at a constant rate defined by SF_FREQ after SF_Init has been called once.
  */ 
-SFError_t SF_GenerateNewResult();
+struct SFError_t SF_GenerateNewResult();
 
 /**
  * Get latest fused sensor data. Can be called any time data is needed after SF_init has been called once. Waits until the output struct is not being accessed by another task.
  * @param [out] output Output struct for fused data.
  * @return Error struct.
  */ 
-SFError_t SF_GetResult(SFOutput_t *SFoutput);
+struct SFError_t SF_GetResult(struct SFOutput_t *SFoutput);
 
 /**
  * Get raw IMU data. Can be called any time raw data is needed.
  * @return IMU struct.
  */ 
-IMU_Data_t SF_GetRawIMU();
+struct IMU_Data_t SF_GetRawIMU();
 
 /**
  * Get raw Airspeed data. Can be called any time raw data is needed.
  * @return Airspeed struct.
  */ 
-Airspeed_Data_t SF_GetRawAirspeed();
+struct Airspeed_Data_t SF_GetRawAirspeed();
```

Comment:
it is no longer necessary to drag the struct keyword around in c++ 11. Why did you put them in. Was something failling ?

Reply:
I think since the functions in sensorfusion are called by freertos they expect it to be in C-Style, because it was failing without the struct keyword

Reply:
hm, I see. This should be taken care of as part of the interface between our functions and FREERTOS. It needs to be, otherwise you'l lbe dissallowing Tony from using C++.